### PR TITLE
Let's Encrypt blog post

### PR DESCRIPTION
**Pull request by ArturT**

### CODE REVIEW

**Comment by ArturT** [comment]
@natif @Tomabana Ready for review.

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+tags:
+  - SSL
+  - TLS
+  - certificate
+  - Rails
+  - https
+  - Let's Encrypt
+---
+
+Recently I was working on adding https to my website [KnapsackPro.com](https://knapsackpro.com) and I'd like to share some tips with you how to configure SSL/TLS in rails application for free with Let's Encrypt.
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: https - upercase maybe?
rails - uppercase maybe?

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+
+<img src="/images/blog/posts/setting-up-automated-ssl-for-rails-app-with-lets-encrypt/lets_encrypt.png" style="float:right;width:300px;" />
+
+I needed a secure connection for my API rails application because my gem called [knapsack_pro](https://github.com/KnapsackPro/knapsack_pro-ruby), which is responsible for optimizing test suite split, sends test file names to API where the test suite split is happening. I wanted to keep connection more secure with SSL/TLS.
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: for my Rails application API
test suite split happens.
to keep the connection

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+
+I was looking for options like maybe buying a cheap certificate for a year but I needed a few certificates for domains like main website, api domain, staging website and api staging domain.
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: like maybe buying - without maybe
for my other domains
api domain - API domain
api staging domain - API staging domain

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+
+A while ago I read on hacker news about [Let's Encrypt](https://letsencrypt.org). It's is a new Certificate Authority sponsored by many companies. They are aiming for a few things:
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: ago,
I read an article

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+
+* free certificates for everyone
+* ssl renewal process should be automated (no more buying a certificate every year and manually updating it on the server)
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: SSL

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+* automatic issuance and renewal protocol as an open standard
+
+What differentiates Let's Encrypt from other Certificate Authorities is that Let's Encrypt has ninety-day lifetimes for certificates. One of the reasons of ninety-day lifetimes is that it encourage automation. We can’t continue to expect system administrators to manually handle renewals. More explanation you can [find here](https://letsencrypt.org/2015/11/09/why-90-days.html).
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: encourages
more explanation can be found/you can find more detailed explanation

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+
+# What you are going to learn
+
+In this article I'm going to show you how to:
+
+* create capistrano tasks to:
+  * register Let's Encrypt client
+  * authorize domain on Let's Encrypt
+  * obtain a certificate from Let's Encrypt
+* create rake task for certificate renewal process and how to run it via cron
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: and run it via cron

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 8]
```diff
@@ -0,0 +1,396 @@
...
+gem 'acme-client', require: false
+{% endhighlight %}
+
+Remember to run `bundle install` after that.
+
+# Capistrano task to register Let's Encrypt client
+
+We have to create an account on Let's Encrypt in order to authenticate our client. The capistrano task will handle that and create a new private key which will be registered on Let's Encrypt.
+
+When you will try to run the task for the second time it will skip registration process because the private key exists.
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: when you try to run

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 9]
```diff
@@ -0,0 +1,396 @@
...
+        # You may need to agree to the terms of service
+        # (that's up the to the server to require it or not but boulder does by default)
+        registration.agree_terms
+      end
+    end
+  end
+end
+{% endhighlight %}
+
+We also need to ensure capistrano gem can see our task. Add below the line at the end of your `Capfile` if the line is missing.
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: ensure that
Add the line listed below at the end

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 10]
```diff
@@ -0,0 +1,396 @@
...
+      info "Creating symlinks for existing certificate."
+      # #{shared_path}/ssl_private_key.key is a path where nginx is looking for ssl certificate
+      sudo "ln -nfs #{cert_privkey_path} #{shared_path}/ssl_private_key.key"
+      sudo "ln -nfs #{cert_fullchain_path} #{shared_path}/ssl_cert.crt"
+    end
+  end
+end
+{% endhighlight %}
+
+Please note that I'm using nginx and [my nginx configuration](https://github.com/KnapsackPro/capistrano-cookbook/blob/master/lib/capistrano/cookbook/templates/nginx.conf.erb#L54,L55) is looking for ssl cert and ssl private key in shared directory. In your case, it might be a different directory. You need to ensure your server like nginx or apache has enabled SSL and specified the path where to look for the certificate.
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: SSL
Apache

File: _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md [hunk 11]
```diff
@@ -0,0 +1,396 @@
...
+{% highlight ruby %}
+# config/deploy.rb
+after 'deploy:symlink:release', 'letsencrypt:register_client'
+after 'letsencrypt:register_client', 'letsencrypt:authorize_domain'
+after 'letsencrypt:authorize_domain', 'letsencrypt:obtain_certificate'
+after 'letsencrypt:obtain_certificate', 'nginx:reload'
+{% endhighlight %}
+
+We need to reload nginx server after we obtain the certificate.
+I assume you have the task like `nginx:reload` or something similar for another http server like apache.
```
Comment (natif) [inline _posts/2016-08-06-setting-up-automated-ssl-for-rails-app-with-lets-encrypt.md]: HTTP
Apache

**Pull request merged**